src/include/ipc/channel_map.h

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Maka <marcin.maka@linux.intel.com>
+//         Slawomir Blauciak <slawomir.blauciak@linux.intel.com>
+
+#ifndef __SOF_AUDIO_CHANNEL_MAP_H__
+#define __SOF_AUDIO_CHANNEL_MAP_H__
+
+#include <ipc/header.h>
+#include <stdint.h>
+
+/**
+ * \brief Channel map, specifies transformation of one-to-many or many-to-one.
+ *
+ * In case of one-to-many specifies how the output channels are computed out of
+ * a single source channel,
+ * in case of many-to-one specifies how a single target channel is computed
+ * from a multichannel input stream.
+ *
+ * Channel index specifies position of the channel in the stream on the 'one'
+ * side.
+ *
+ * Ext ID is the identifier of external part of the transformation. Depending
+ * on the context, it may be pipeline ID, dai ID, ...
+ *
+ * Channel mask describes which channels are taken into account on the "many"
+ * side. Bit[i] set to 1 means that i-th channel is used for computation
+ * (either as source or as a target).
+ *
+ * Channel mask is followed by array of coefficients in Q2.30 format,
+ * one per each channel set in the mask (left to right, LS bit set in the
+ * mask corresponds to ch_coeffs[0]).
+ */
+struct sof_ipc_channel_map {
+	uint32_t ch_index;
+	uint32_t ext_id;
+	uint32_t ch_mask;
+	int32_t ch_coeffs[0];
+};
+
+/**
+ * \brief Complete map for each channel of a multichannel stream.
+ *
+ * num_ch_map Specifies number of items in the ch_map.
+ * More than one transformation per a single channel is allowed (in case
+ * multiple external entities are transformed).
+ * A channel may be skiped in the transformation list, then it is filled
+ * with 0's by the transformation function.
+ */
+struct sof_ipc_stream_map {
+	struct sof_ipc_cmd_hdr hdr;
+	uint32_t num_ch_map;
+	struct sof_ipc_channel_map ch_map[0];
+};
+
+/** Maximum number of unique physical links in one-to-many configuration */
+#define CHANNEL_MAP_MAX_LINKS	4
+
+/** Maximum number for channel transformations in the map */
+#define CHANNEL_MAP_MAX_ELEMS	8
+
+/* Returns a channel map valid for a given index in a given stream map */
+struct sof_ipc_channel_map *get_channel_map(struct sof_ipc_stream_map *smap,
```

Comment:
It's weird, that we have function prototype in the ipc header shared with the kernel. It's not like it can be used anyway.

Reply:
Yep, this should not be in IPC header.